packages/vite/src/module-runner/utils.ts

```diff
@@ -16,7 +16,7 @@ export function normalizeAbsoluteUrl(url: string, root: string): string {
     // /root/id.js -> /id.js
```

Comment:
I guess the addition of `resolveId` in `fetchModule` makes this irrelevant. The only downside is the additional call to the server if the entry point is not normalized, but it was already imported. 

If we move this to the environment, then maybe this whole function can be removed, which also makes it so the `root` is not required anymore

Reply:
Okay, I moved everything to `fetchModule` including `file://` handling. It looks like this works, but the behavior of `file://` might be actually different from browser and v5 since Astro is relying on userland plugin https://github.com/vitejs/vite/pull/17369#issuecomment-2306733108